crates/ra_lsp_server/src/main_loop/handlers.rs

```diff
@@ -467,27 +465,16 @@ pub fn handle_rename(world: ServerWorld, params: RenameParams) -> Result<Option<
         .into());
     }
 
-    let renames = world
+    let change = world
         .analysis()
         .rename(FilePosition { file_id, offset }, &*params.new_name)?;
-    if renames.is_empty() {
+    if change.is_none() {
         return Ok(None);
     }
 
-    let mut changes = HashMap::new();
-    for edit in renames {
-        changes
-            .entry(file_id.try_conv_with(&world)?)
-            .or_insert_with(Vec::new)
-            .extend(edit.edit.conv_with(&line_index));
-    }
-
-    Ok(Some(WorkspaceEdit {
-        changes: Some(changes),
+    let source_change_req = change.unwrap().try_conv_with(&world)?;
```

Comment:
I think this `.unwrap()` may actually panic? I think we should return an `Ok(None)` if the change is `None`?